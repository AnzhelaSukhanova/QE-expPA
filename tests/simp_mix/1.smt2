(set-option :produce-models true)
(set-logic BV)
(declare-fun y () (_ BitVec 4))
(declare-fun z () (_ BitVec 4))
(assert
 (exists ((x (_ BitVec 4))) (and
 (or (bvult (bvshl #b0001 x) z) (= (bvshl #b0001 x) z))
 (or (bvult x y) (= x y))))) 
