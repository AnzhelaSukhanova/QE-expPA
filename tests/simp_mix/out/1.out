(let (($e1 (= z #x0))) (let (($e2 (ite (bvult z #x8) (ite (bvult z #x4) (ite (bvult z #x2) (ite $e1 #xf #x0) #x1) #x2) #x3))) (let (($e3 (= z #x1))) (let (($e4 (and (not (= y #x1)) (not (bvult #x1 y))))) (let (($e5 (and (not $e4) (= z #x2)))) (let (($e6 (and (not (= y #x2)) (not (bvult #x2 y))))) (let (($e7 (and (not $e6) (= z #x4)))) (let (($e8 (and (not (= y #x3)) (not (bvult #x3 y))))) (let (($e9 (and (not $e8) (= z #x8)))) (let (($e10 (and (not (= y #x4)) (not (bvult #x4 y))))) (let (($e11 (and (not $e10) $e1))) (let (($e12 (and (not (= y #x5)) (not (bvult #x5 y))))) (let (($e13 (and (not $e12) $e1))) (let (($e14 (and (not (= y #x6)) (not (bvult #x6 y))))) (let (($e15 (and (not $e14) $e1))) (let (($e16 (and (not (= y #x7)) (not (bvult #x7 y))))) (let (($e17 (and (not $e16) $e1))) (let (($e18 (and (not (= y #x8)) (not (bvult #x8 y))))) (let (($e19 (and (not $e18) $e1))) (let (($e20 (and (not (= y #x9)) (not (bvult #x9 y))))) (let (($e21 (and (not $e20) $e1))) (let (($e22 (and (not (= y #xa)) (not (bvult #xa y))))) (let (($e23 (and (not $e22) $e1))) (let (($e24 (and (not (= y #xb)) (not (bvult #xb y))))) (let (($e25 (and (not $e24) $e1))) (let (($e26 (and (not (= y #xc)) (not (bvult #xc y))))) (let (($e27 (and (not $e26) $e1))) (let (($e28 (and (not (= y #xd)) (not (bvult #xd y))))) (let (($e29 (and (not $e28) $e1))) (let (($e30 (and (not (= y #xe)) (not (bvult #xe y))))) (let (($e31 (and (not $e30) $e1))) (let (($e32 (= y #xf))) (let (($e33 (and $e32 $e1))) (let (($e34 (bvult #x1 z))) (let (($e35 (and (bvult #x2 z) (not $e4)))) (let (($e36 (and (bvult #x4 z) (not $e6)))) (let (($e37 (and (bvult #x8 z) (not $e8)))) (let (($e38 (and (not $e10) (not $e1)))) (let (($e39 (and (not $e12) (not $e1)))) (let (($e40 (and (not $e14) (not $e1)))) (let (($e41 (and (not $e16) (not $e1)))) (let (($e42 (and (not $e18) (not $e1)))) (let (($e43 (and (not $e20) (not $e1)))) (let (($e44 (and (not $e22) (not $e1)))) (let (($e45 (and (not $e24) (not $e1)))) (let (($e46 (and (not $e26) (not $e1)))) (let (($e47 (and (not $e28) (not $e1)))) (let (($e48 (and (not $e30) (not $e1)))) (let (($e49 (and $e32 (not $e1)))) (not (and (not $e35) (not $e34) (not $e36) (not $e5) (not $e3) (not $e37) (not $e7) (not $e38) (not $e9) (not $e39) (not $e11) (not $e40) (not $e13) (not $e41) (not $e15) (not $e42) (not $e17) (not $e43) (not $e19) (not $e44) (not $e21) (not $e45) (not $e23) (not $e46) (not $e25) (not $e47) (not $e27) (not $e48) (not $e29) (not $e49) (not $e31) (not $e34) (not $e33) (not $e35) (not $e3) (not $e36) (not $e5) (not $e37) (not $e7) (not $e38) (not $e9) (not $e39) (not $e11) (not $e40) (not $e13) (not $e41) (not $e15) (not $e42) (not $e17) (not $e43) (not $e19) (not $e44) (not $e21) (not $e45) (not $e23) (not $e46) (not $e25) (not $e47) (not $e27) (not $e48) (not $e29) (not $e49) (not $e31) (not $e34) (not $e33) (not $e35) (not $e3) (not (and (not (and (not (= y $e2)) (not (bvult $e2 y)))) (bvult (bvshl #x1 $e2) z))) (not $e5))))))))))))))))))))))))))))))))))))))))))))))))))))
